Livraison 2 : Tâche Guiconf , afficher une carte avec le lieu de la conférence

feLlx opened the pull request:
Cette PR contient les changements effectués pour afficher à l'utilisateur une carte avec le lieu de sa conférence :
- ajout d'un bouton "See location conference" qui affiche la carte avec un point qui a pour coordonnées la latitude et longitude la ville écrite dans le champ "city" (et si la ville est dans le nouveau fichier txt "cities15000").
- la latitude et la longitude la ville d'arrivée sont stockés dans une instance de la classe PathStep qui a été modifié.
- les fichiers de tests concernant les classes modifiées ont été modifiés également pour les transformer en tests Junit.
Dans une future livraison, on pourra tracer un chemin entre le point de départ et d'arrivée ainsi que la longueur qui les sépare.

Commit:
Conference location

Commit:
Junit test after modif GeoCode and use of map

Commit:
Map path with class PathStep

Commit:
Change listener & delete warnings

Commit:
Indentation

oliviercailloux commented on src/main/java/io/github/oliviercailloux/y2018/geocode/ReverseGeoCode.java:
```diff
@@ -80,7 +85,7 @@ public ReverseGeoCode( InputStream placenames, boolean majorOnly ) throws IOExce
     private void createKdTree(InputStream placenames, boolean majorOnly)
             throws IOException {
-        ArrayList<GeoName> arPlaceNames;
+        //ArrayList<GeoName> arPlaceNames;
```
Éviter le code désactivé. On peut revenir en arrière grâce à git, c’est à ça que ça sert.

feLlx replied:
Ligne supprimée

oliviercailloux reviewed (changes requested):
FORMAT (je continue la review après).

feLlx commented:
 Itération 3

Commit:
Add logger

Commit:
ajout fichier pdf recapitulatif recherche api geolocalisation maps it 3 -> 4

Commit:
Merge conflict + add file map help + use of reference

Commit:
Travis resolve

Commit:
merge conflicts

feLlx commented:
Itération 4 commencée depuis le 20/06 :

Suite aux difficultés rencontrées lors de l'implémentation d'une carte et le tracer d'un itinéraire, nous avons rédigé une documentation contenant nos recherches et nos recommandations pour une meilleure compréhension des API proposées.

Dans le code des classes "GuiConference" et "TestGeocode", nous avons masqué le warning "Potential resource leak: '<unassigned Closeable value>' may not be closed" avec des @SuppressWarnings("resource"). La raison est parce qu'il est impossible de faire un try-with-ressources (qui résoudrait le warning) à cause de la classe Geocode qui n'implémente pas java.lang.AutoCloseable et ne peut donc être fermé.

Il y a dix fichiers modifiés dans cette pr : 
- deux documents pdf pour aider le prochain groupe
- deux classes de tests que nous avons transformé en vrai test
- deux fichiers txt contenant des villes
- les classes ReverseGeoCode et GeoName contenant de légères modifications pour avoir accès aux informations des villes des fichiers txt
- la classe PathStep modifié pour pouvoir stocker les informations des villes, en particulier, la latitude et longitude
- la classe GuiConference utilisé pour afficher une carte avec le lieu de la conférence

Commit:
add javadoc

Commit:
indentation

Commit:
solve travis

oliviercailloux commented on src/main/java/io/github/oliviercailloux/jconfs/gui/GuiConference.java:
```diff
@@ -399,6 +408,64 @@ public void generateYs(@SuppressWarnings("unused") Event e) {
+		 * (For instance If you look for "Pekin" in french, you have to search "Beijing".*/
+		ReverseGeoCode reverseGeoCode;
+		try {
+			reverseGeoCode = new ReverseGeoCode(new FileInputStream("src/main/resources/io/github/oliviercailloux/jconfs/map/cities15000.txt"), true);
```
BP resources

feLlx replied:
Application de la bp

oliviercailloux commented on src/test/java/TestGeocode.java:
```diff
@@ -10,13 +14,20 @@
  *
  */
 public class TestGeocode {
```
Placez cette classe dans un package correct, tant que vous y êtes.

feLlx replied:
Les classes tests ont été réparties dans des packages corrects

oliviercailloux commented on src/test/java/io/github/oliviercailloux/TestCalendarOnline.java:
```diff
@@ -38,7 +38,6 @@ public void testGetOnlineConferenceFromUid() throws CalDAV4JException, InvalidCo
 		assertEquals(conferenceFound.getFeeRegistration().toString(), "1.46");
 	}
 
-	@Test
```
Pourquoi ?

feLlx replied:
Suppression pour ne pas avoir d'erreurs avec le build Travis. Test non réalisé par moi même

oliviercailloux commented on src/test/java/io/github/oliviercailloux/TestMap.java:
```diff
@@ -11,14 +14,14 @@
+	public void openMapTest() throws FileNotFoundException{
+		String end;
+		URL fileURL = TestMap.class.getClassLoader().getResource("io/github/oliviercailloux/jconfs/map/world.map");
+		System.out.println(fileURL);
```
Non, utiliser un logger.

feLlx replied:
syso supprimé

oliviercailloux commented on src/test/java/io/github/oliviercailloux/TestMap.java:
```diff
@@ -29,8 +32,9 @@ public static void main(String[] args) throws FileNotFoundException {
-		System.out.println("done");
+		end = "done";
 		map.close();
+		assertEquals(end,"done");
```
- BP resources : une ressource dans un jar n’est pas convertible en Fichier. Il faut se contenter d’un stream.
- Votre test devrait tester le contenu de la carte. (Tester que `end` égale `"done"` juste après l’avoir affecté à `"done"`, c’est se moquer de votre lecteur.)

feLlx replied:
Ce test a été réalisé par l'ancien groupe. Ce n'était même pas un test à la base. Je l'ai uniquement transformé en test. Si quelqu'un se moque du lecteur, je ne pense pas que cela soit moi.

feLlx replied:
@oliviercailloux je ne comprends votre remarque première remarque.

feLlx replied:
Utilsation de get.Path après application de la bp ressource

oliviercailloux replied:
- Ne pas accepter `File` en paramètre mais un `InputStream`, plus général : certains flux ne correspondent pas à des fichiers
- L’ancien groupe s’est lui aussi moqué du lecteur, je vous l’accorde (et bien plus que vous). C’est certes une circonstance atténuante, mais ce n’est pas une très bonne excuse. Si vous affirmez transformer leur méthode en un test, il faut qu’il teste quelque chose. Si vous ne désirez pas le faire, alors c’est erroné de prétendre le faire. Dans ce deuxième cas, je suggère de simplement indiquer un commentaire en Javadoc indiquant TODO make a real test out of this. En tous cas, vous pouvez effacer le `assertEquals` qui ne sert à rien, pour ne pas laisser croire erronément que ce test teste effectivement quelque chose.

oliviercailloux commented on src/main/java/io/github/oliviercailloux/geocode/ReverseGeoCode.java:
```diff
@@ -46,6 +46,11 @@ of this software and associated documentation files (the "Software"), to deal
 	KDTree<GeoName> kdTree;
+	ArrayList<GeoName> arPlaceNames;
+
+	public ArrayList<GeoName> getArPlaceNames() {
```
Non, renvoyer `List`, plus général.

feLlx replied:
Transformation effectuée.

oliviercailloux commented on src/main/java/io/github/oliviercailloux/jconfs/gui/GuiConference.java:
```diff
@@ -399,6 +408,64 @@ public void generateYs(@SuppressWarnings("unused") Event e) {
+				mapGUI.display();				
+
+			} catch (NullPointerException | IllegalArgumentException | IOException e1) {
+				e1.printStackTrace();
```
EXC

feLlx replied:
throw new IllegalStateException(e1);

oliviercailloux commented on src/main/java/io/github/oliviercailloux/jconfs/gui/GuiConference.java:
```diff
@@ -399,6 +408,64 @@ public void generateYs(@SuppressWarnings("unused") Event e) {
+				}
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
```
EXC

feLlx replied:
throw new RuntimeException(e);

oliviercailloux commented on src/main/java/io/github/oliviercailloux/jconfs/map/PathStep.java:
```diff
@@ -93,10 +150,58 @@ public void setArrivalPoint(String arrivalPoint) {
+	 *            not <code>null</code>.
+	 */
+	public void setLatitudeStartingPoint(double latitudeStartingPoint) {
+		this.latitudeStartingPoint = Objects.requireNonNull(latitudeStartingPoint);
```
Type primitif, ne peut être null

feLlx replied:
Suppression

oliviercailloux reviewed (changes requested):
- FORMAT
- Documents demandés au format Asciidoctor, pour modification, commentaires, réutilisation plus simple.
- Vous ne pouvez pas utiliser try-with-resource, mais cela ne devrait pas vous empêcher de fermer les ressources ouvertes en appelant `close()`. En tous cas, choisir de les laisser simplement ouvertes n’est pas une bonne solution.
- Utilisez-vous AU.txt (et autres) pour autre chose que les tests ?
- Utiliser un objet qui représente un point sur une carte (avec latitude, longitude et nom) dans `PathStep`, j’imagine que vous avez ceci sous la main.

Commit:
pdf -> adoc

Commit:
package for tests

Commit:
no need of objects.requirenonnull

Commit:
exception

Commit:
arraylist -> list

Commit:
delete AU.txt + application of bp ressources

Commit:
Change of TestMap

jacquesmoati committed:
add readme Document : Indications et recherche sur les API de cartes

Commit:
Application bp ressources

Commit:
Merge branch 'map' of https://github.com/13tomoore/J-Confs into map

feLlx commented:
Fichier AU.txt supprimé
25 fichiers changés en raison de la création et du déplacement de package pour appliquer la bp ressources

jacquesmoati committed:
add readme images

jacquesmoati committed:
Update API_MAP_Indication.adoc

real path

Commit:
file -> stream

Commit:
Merge branch 'map' of https://github.com/13tomoore/J-Confs into map

jacquesmoati commented:
@oliviercailloux 
API_MAP_Indication.adoc ce fichier contient toute la doc de recherche et nécessaire pour un travail ultérieur sur les maps dans le projet J-Confs. Comme demandé en asciidoc. 
PS : Je n'arrive pas à afficher clairement dans le readme les images .png  avec le path. J'ai pourtant respecté le path. 
Exemple
![alt text](https://github.com/13tomoore/J-Confs/blob/map/Doc/Itin%C3%A9raire_Explication.png)

oliviercailloux commented on src/main/java/io/github/oliviercailloux/geocode/GeoName.java:
```diff
@@ -47,6 +47,18 @@ of this software and associated documentation files (the "Software"), to deal
 	public double point[] = new double[3]; // The 3D coordinates of the point
 	public String country;
+	
+	public String getName() {
```
Ne modifiez pas cette classe qui ne nous appartient pas SVP ! On veut utiliser ce code comme une bibliothèque. Tant pis s’il est imparfait. Les champs étant publics, vous y avez accès.

feLlx replied:
Suppression des getters

oliviercailloux commented on src/main/java/io/github/oliviercailloux/geocode/ReverseGeoCode.java:
```diff
@@ -46,6 +47,11 @@ of this software and associated documentation files (the "Software"), to deal
  */
 public class ReverseGeoCode {
 	KDTree<GeoName> kdTree;
+	List<GeoName> arPlaceNames;
```
Même chose.

feLlx replied:
suppression du getter

oliviercailloux commented on src/main/java/io/github/oliviercailloux/jconfs/map/PathStep.java:
```diff
@@ -14,6 +14,10 @@
 	private TransportType type = TransportType.NOTRANSPORT;
 	private String startingPoint = "";
 	private String arrivalPoint = "";
+	private double latitudeStartingPoint;
```
Cf. commentaire précédent.

feLlx replied:
Création d'une classe Point ajoutée

oliviercailloux commented on src/test/java/io/github/oliviercailloux/jconfs/map/TestGeocode.java:
```diff
@@ -0,0 +1,33 @@
+	public void nearestPlaceTest() throws FileNotFoundException, IOException {
+		URL resourceUrl = GuiConference.class.getResource("cities15000.txt");
+		ReverseGeoCode reverseGeoCode = new ReverseGeoCode(resourceUrl.openStream(), true);
+		assertEquals(reverseGeoCode.nearestPlace(39.913818, 116.363625).getName(),"Beijing");
```
Paramètres dans le bon ordre SVP.

feLlx replied:
changement de l'ordre

oliviercailloux reviewed (changes requested):
- FORMAT
- Doc au format asciidoctor doit apparaitre correctement sur GitHub. Pour répondre à votre question : êtes-vous sûr que vous essayez d’utiliser le format asciidoctor et non Markdown ?

Commit:
public attribute

Commit:
order parameters

Commit:
class point in PathStep

Commit:
format

Commit:
correction errors

Commit:
modify adoc

jacquesmoati committed:
Update API_MAP_Indication.adoc

petite modification

feLlx commented:
Les documents adoc ont été réalisé avec le plugin suivant : https://marketplace.eclipse.org/content/asciidoctor-editor?fbclid=IwAR12SLtbYHnRfoBpDNxl4SVnDjW4hWzDpx-H7Gg4qlPBODs4LeeBMns-Jjs

Modifcation de la classe PathStep
Modification de geocode (suppression des ajouts)

jacquesmoati committed:
Update API_MAP_Indication.adoc

Résolution des problèmes d'affichage d'images

jacquesmoati commented:
@oliviercailloux 
Effectivement la syntaxe n'était pas bonne. J'ai pu modifier, les images se sont bien affichées dans le fichier API_MAP_Indication.adoc .

oliviercailloux commented on Doc/API_MAP_Indication.adoc:
Ce code json n’apparait pas correctement sur GitHub, comme vous pouvez le constater.

feLlx replied:
Correction apportée

oliviercailloux reviewed (changes requested):
Veuillez restaurer la classe `ReverseGeoCode` telle qu’elle était précédemment, pour la même raison.

Commit:
Not public

Commit:
Merge branch 'map' of https://github.com/13tomoore/J-Confs into map

feLlx commented:
Si les attributs de la classe ReverseGeoCode ne sont pas publics, mon code ne marche plus. Vous me demandez des modifications pour ensuite me dire d'annuler. Je ne comprends pas. Mais soit, les attributs ne sont plus publics et j'ai dû mettre ce que j'avais développé en commentaire. Le prochaine groupe s'en chargera.

Commit:
merge conflicts

oliviercailloux commented:
J’imagine qu’il y a malentendu, car je ne comprends pas où vous voyez une annulation d’un changement demandé précédemment. 

Mon commentaire à propos de la classe `GeoName` était le suivant : « Ne modifiez pas cette classe qui ne nous appartient pas SVP ! On veut utiliser ce code comme une bibliothèque. Tant pis s’il est imparfait. » Vous avez correctement réagi en restaurant cette classe telle qu’elle était originalement. Je faisais simplement valoir que ce commentaire s’applique également (évidemment) sur `ReverseGeoCode` et qu’il faut donc la restaurer également telle qu’elle était originalement.

Commit:
original RGeoCode

Commit:
mute test

feLlx commented:
Retour à l'original (j'ai juste gardé les corrections pour ne pas avoir de warnings).

jacquesmoati committed:
Update API_MAP_Indication.adoc

@oliviercailloux J'ai corrigé avec une indentation pour mieux visualiser le contenu du JSON

jacquesmoati commented:
@oliviercailloux J'ai corrigé avec une indentation pour mieux visualiser le contenu du JSON

feLlx commented:
@oliviercailloux

oliviercailloux commented on src/main/java/io/github/oliviercailloux/jconfs/map/PathStep.java:
```diff
@@ -11,9 +11,66 @@
  */
 public class PathStep {
 
+	public class Point{
```
Il aurait été plus simple de réutiliser `GeoName` comme suggéré dans mon commentaire. 

Si vous tenez à une nouvelle classe, évitez d’imbriquer une classe dans une autre, faites-en une vraie classe (dans son propre fichier) pour plus de lisibilité, et nommez-la par exemple `GeoPoint` pour plus de clarté. 

Et, facultatif, faites-en une classe immuable.

feLlx replied:
Immuable classe GeoPoint créée

oliviercailloux commented on src/main/java/io/github/oliviercailloux/jconfs/gui/GuiConference.java:
```diff
@@ -399,6 +410,66 @@ public void generateYs(@SuppressWarnings("unused") Event e) {
+	 *            not <code>null</code>.
+	 * @return 
+	 */
+	public static PathStep getLatLonCity(String city) {
```
Vous me semblez très près du but, à moins qu’une subtilité m’échappe. Ce serait dommage de renoncer maintenant.

Pourquoi ne pas simplement copier une partie de la méthode `createKdTree` ici (en indiquant bien sûr en commentaire la provenance de votre copie) ? Si je comprends bien, c’est tout ce dont vous avez besoin pour faire fonctionner votre code, et cela évite de modifier la bibliothèque geoname.

Sinon, il faut appliquer BP style, effacer cette méthode `getLatLonCity` pour éviter que qqn l’utilise par erreur, et indiquer dans votre README l’identifiant du commit contenant votre code pour permettre aux successeurs de le terminer.

feLlx replied:
Utilisation du code de la méthode createKdTree et du code du constructeur de GeoName. Mention du code d'origine et du créateur.

oliviercailloux reviewed (changes requested):
Veuillez restaurer la classe `ReverseGeoCode` telle qu’elle était exactement, pour éviter des différences dans l’historique qui peuvent créer des interrogations sur une possible modification de la bibliothèque lors d’une mise à jour éventuelle.

Commit:
back to original

Commit:
Immuable class GeoPoint

Commit:
work on getLatlon

Commit:
mute test

Commit:
Test

oliviercailloux commented on src/main/java/io/github/oliviercailloux/jconfs/gui/GuiConference.java:
```diff
@@ -448,26 +452,41 @@ public void displayMap(@SuppressWarnings("unused") Event e) {
-		/*URL resourceUrl = GuiConference.class.getResource("cities15000.txt");
-		ReverseGeoCode reverseGeoCode;
+		URL resourceUrl = GuiConference.class.getResource("cities15000.txt");
+		/*line 461 to 476 are taken from ReverseGeoCode.java and GeoName.java classes with a few modifications
```
Les n° de lignes vont très rapidement devenir incorrects sans que personne ne s’en rende compte. Il faut délimiter le code autrement, par exemple par un bloc.

feLlx replied:
Délimitation avec un bloc.

oliviercailloux reviewed (approved):
FORMAT

Commit:
delimitation block

nikolaspaci commented:
PR approuvée, merge dans master de la dernière tâche de l'itération 4
